diff --git a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFactory.java b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFactory.java
index 2e9d756..a8339ed 100644
--- a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFactory.java
+++ b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleFactory.java
@@ -3,6 +3,9 @@
 
 package com.mapbox.rctmgl.components.styles;
 
+import android.util.Log;
+
+import com.mapbox.mapboxsdk.style.expressions.Expression;
 import com.mapbox.mapboxsdk.style.layers.BackgroundLayer;
 import com.mapbox.mapboxsdk.style.layers.CircleLayer;
 import com.mapbox.mapboxsdk.style.layers.FillExtrusionLayer;
@@ -1614,7 +1617,12 @@ public class RCTMGLStyleFactory {
       if (styleValue.isExpression()) {
         layer.setProperties(PropertyFactory.circleRadius(styleValue.getExpression()));
       } else {
-        layer.setProperties(PropertyFactory.circleRadius(styleValue.getFloat(VALUE_KEY)));
+          try {
+              layer.setProperties(PropertyFactory.circleRadius(styleValue.getFloat(VALUE_KEY)));
+          } catch (Exception e) {
+              layer.setProperties(PropertyFactory.circleRadius(0f));
+//              layer.setProperties(PropertyFactory.circleRadius(Expression.interpolate(Expression.exponential(2), Expression.zoom(), Expression.stop(0, 1),Expression.stop(10, 100))));
+          }
       }
     }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleValue.java b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleValue.java
index f356032..97e6291 100644
--- a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleValue.java
+++ b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/components/styles/RCTMGLStyleValue.java
@@ -72,6 +72,10 @@ public class RCTMGLStyleValue {
                     mExpression = ExpressionParser.fromTyped(mPayload);
                 }
             }
+//            else if (mPayload.getString("type").equals("hashmap")) {
+//                isExpression = true;
+//                mExpression = ExpressionParser.fromTyped(mPayload);
+//            }
         }
     }
 
diff --git a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ConvertUtils.java b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ConvertUtils.java
index 3c739dd..98a2a9f 100644
--- a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ConvertUtils.java
+++ b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ConvertUtils.java
@@ -95,6 +95,7 @@ public class ConvertUtils {
         String type = map.getString("type");
 
         if (type.equals(ExpressionParser.TYPE_MAP)) {
+//            JsonArray resultArr = new JsonArray(1);
             JsonObject result = new JsonObject();
 
             ReadableArray keyValues = map.getArray("value");
@@ -104,6 +105,7 @@ public class ConvertUtils {
 
                 result.add(key, typedToJsonElement(keyValue.getMap(1)));
             }
+//            resultArr.add(result);
             return result;
         }
         else if (type.equals(ExpressionParser.TYPE_ARRAY)) {
diff --git a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ExpressionParser.java b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ExpressionParser.java
index d63aa51..2097452 100644
--- a/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ExpressionParser.java
+++ b/node_modules/@react-native-mapbox-gl/maps/android/rctmgl/src/main/java/com/mapbox/rctmgl/utils/ExpressionParser.java
@@ -33,8 +33,6 @@ public class ExpressionParser {
         return Expression.Converter.convert(array);
     }
 
-
-
     public static Expression from(ReadableMap rawExpression) {
         return Expression.raw("[" + stringExpression(rawExpression) + "]");
     }
diff --git a/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLStyle.m b/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLStyle.m
index 6723bac..7565d8e 100644
--- a/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLStyle.m
+++ b/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/RCTMGLStyle.m
@@ -353,7 +353,11 @@
     if ([prop isEqualToString:@"visibility"]) {
       [self setCircleStyleLayerVisibility:layer withReactStyleValue:styleValue];
     } else if ([prop isEqualToString:@"circleRadius"]) {
-      [self setCircleRadius:layer withReactStyleValue:styleValue];
+      @try {
+        [self setCircleRadius:layer withReactStyleValue:styleValue];
+      }
+      @catch (NSException *e) {
+      }
     } else if ([prop isEqualToString:@"circleRadiusTransition"]) {
       [self setCircleRadiusTransition:layer withReactStyleValue:styleValue];
     } else if ([prop isEqualToString:@"circleColor"]) {
diff --git a/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/index.d.ts b/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/index.d.ts
index 3c19b6a..01349f7 100644
--- a/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/index.d.ts
+++ b/node_modules/@react-native-mapbox-gl/maps/ios/RCTMGL/index.d.ts
@@ -685,7 +685,7 @@ export interface circleLayerStyleProps {
   /**
    * Circle radius.
    */
-  circleRadius: number;
+  circleRadius: number | StyleFunctionProps;
 
   /**
    * The transition affecting any changes to this layer’s circleRadius property.
